### PR TITLE
[dtensor][math_ops]  Add single_dim_strategy functions for sort-like, scan and softmax ops

### PR DESCRIPTION
**Pull request by anshul-si**
**Before**
op_strategy (register_op_strategy): 145
single_dim_strategy: 1022
Decomposition table (not directly registered): 435
 
**After**
op_strategy (register_op_strategy): 132
single_dim_strategy: 1049
Decomposition table (not directly registered): 427

6 New Ops Added

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #179203
* #179202
* #179201
* #179200
* __->__ #179068

### CODE REVIEW

**Comment by zpcore** [comment]
@claude review PR

**Review by zpcore — APPROVED** [review]
LGTM!

**Review by zpcore — APPROVED** [review]

**Comment by anshul-si** [comment]
@claude is the [pull / linux-jammy-py3.14-clang18 / test-osdc (default, 2, 5, mt-l-x86iavx512-16-128)](https://hud.pytorch.org/pr/pytorch/pytorch/179068#80720149771) ([gh](https://github.com/pytorch/pytorch/actions/runs/27323495788/job/80720149771)) an issue caused by changes in this pr, or unrelated
